### DISCUSSION TITLE
Automate queries of Power BI Service dataset - connection string parameters

### DISCUSSION THREAD

**Discussion by Declan1984** [DaxStudio/DaxStudio#864] [Q&A]
Hi - I'm trying to build automated queries of our Power BI Dataset hosted in the Power BI Service using our ETL tool so that we can export specific measure calculations to other systems.  The key is setting the proper connection string values to establish the connection with the Power BI service.

Obviously, DAX Studio does this to support running OLAP queries and conveniently makes it easy to connect to Power BI Datasets hosted in the Power BI Service with sign in through our enterprise Active Directory so it is all done through connection wizard windows.

However, for my process I would like to see the specific connection settings used by DAX Studio for reference.   Is there a way to see the detailed connection parameters used in DAX Studio when connecting to Power BI Datasets?  Is there some other documentation outside of DAX Studio I could reference that would help set up custom queries against a Power BI Dataset?  Thanks.

**Comment by dgosbell** [comment]
> However, for my process I would like to see the specific connection settings used by DAX Studio for reference. Is there a way to see the detailed connection parameters used in DAX Studio when connecting to Power BI Datasets?

You can see the exact connection that is used. If you hold down the SHIFT key while starting DAX Studio it will create debug level logs and if you connect to a PBI XMLA endpoint the connection string that is used will be written the logs.

I don't do anything special though. I basically use the following format
```
Data Source=<pbi endpoint>;Initial Catalog=<dataset name>
```
You probably will not see the `Initial Catalog=` setting in the original connection, that would get set later based on the Database dropdown in the metadata pane. Your biggest issue with incorporating this into some sort of ETL tools is going to be figuring out the authentication. DAX Studio can prompt the user for credentials, but typically for an ETL flow you will want to use some sort of unattended account.

**Reply by Declan1984** [reply]
Thanks @dgosbell - didn't know about the logging option.  I did, however, need to reset my logging options to Verbose from Warning in order to see the full connection string.

In the connection string, the format is "powerbi://<end point>" which lines up with what I was already trying based on the XMLA endpoint displayed in the dataset settings in Power BI service.  However, my ETL connector was throwing an error that it needs an HTTPS end point.  Is "powerbi://" a universally recognized internet protocol that will route over the web, or is it some type of alias for another URL that is handled "under the hood" in Microsoft based applications?  Not sure if DAX Studio (and Tabular Editor) had Microsoft sourced libraries for handling the URL routing?

**Reply by dgosbell** [reply]
> Is "powerbi://" a universally recognized internet protocol that will route over the web

I don't think so. I think it is a url scheme that is specifically recognised by the Microsoft ADOMD / AMO libraries

> Not sure if DAX Studio (and Tabular Editor) had Microsoft sourced libraries for handling the URL routing?

Yes, both these tools use Microsoft source libraries. These are available on Nuget or you can download them from here [Not sure if DAX Studio (and Tabular Editor) had Microsoft sourced libraries for handling the URL routing? ](https://docs.microsoft.com/en-us/analysis-services/client-libraries?view=sql-analysis-services-2022) 

It almost sounds like your ETL connector is trying to use an old version of the libraries, possibly installing the current release will enable them to recognise the powerbi:// scheme.

**Reply by Declan1984** [reply]
The ETL connector I am using is JDBC based so I don't think it would be possible to install these libraries.  I did find this JDBC Power BI XMLA driver that looks interesting: https://www.cdata.com/drivers/powerbixmla/jdbc/
